src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -582,6 +584,8 @@ public static void PtrToStructure<T>(IntPtr ptr, [DisallowNull] T structure)
             PtrToStructure(ptr, (object)structure!);
         }
 
+        [UnconditionalSuppressMessage("AotAnalysis", "IL9700:AotUnfriendlyApi",
+            Justification = "AOT compilers can see the T.")]
```

Comment:
I am wondering whether it would be better to call `PtrToStructureHelper` instead of suppressing the warning. Some of the argument checks would need to be duplicated, but we would be also able to omit some.

Reply:
(similar to other places like `GetDelegateForFunctionPointer`)